vars/jenkinsMaterializeLog.groovy

```diff
@@ -0,0 +1,80 @@
+import com.sap.piper.ConfigurationHelper
+import com.sap.piper.GenerateDocumentation
+import com.sap.piper.k8s.ContainerMap
+import hudson.FilePath
+import com.sap.piper.Utils
+import groovy.transform.Field
+import java.util.UUID
+import java.io.File
+import com.cloudbees.groovy.cps.NonCPS
+import com.sap.piper.JenkinsUtils
+
+import static com.sap.piper.Prerequisites.checkScript
+
+@Field String STEP_NAME = getClass().getName()
+@Field Set GENERAL_CONFIG_KEYS = []
+
+@Field Set STEP_CONFIG_KEYS = GENERAL_CONFIG_KEYS
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+
+
+/**
+ * This step allows you to materialize the Jenkins log file of the running build
+ */
+@GenerateDocumentation
+void call(Map parameters = [:], body) {
+	handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters)
+	{
+		def jenkinsUtils = parameters.jenkinsUtilsStub ?: new JenkinsUtils()
+		checkScript(this, parameters) ?: this

```

Comment:
`?:` does not make sense since we do not assign to any variable. `checkScript(this, parameters)` is sufficient here. In case the script is not used anywhere it would be able drop even this. Only reason for keeping it nevertheless is the consistency among all the steps (almost all need the reference to the surrounding script).